socorro/unittest/external/es/test_crashstorage.py

```diff
@@ -190,6 +192,84 @@ def test_index_crash_indexable_keys(self):
         processed_crash = doc["_source"]["processed_crash"]
         assert list(sorted(processed_crash.keys())) == ["date_processed", "uuid"]
 
+    def test_index_crash_mapping_keys(self):
+        """Test indexing a crash that has keys not in the mapping
+
+        Indexing a crash that has keys that aren't in the mapping for the index
+        should cause those keys to be removed from the crash.
+
+        """
+        # The test harness creates an index for this week and last week. So let's create
+        # one for 4 weeks ago.
+        now = utc_now()
+        four_weeks_ago = now - timedelta(days=28)
+
+        field = "user_comments"
+
+        # We're going to use a mapping that's what SuperSearchFields gives us, but
+        # remove the user_comments field.
+        mappings = SuperSearchFields(context=self.es_context).get_mapping()
+        doctype = self.es_context.get_doctype()
+        del mappings[doctype]["properties"]["processed_crash"]["properties"][field]
+
+        # Create the index for 4 weeks ago
+        self.es_context.create_index(
+            index_name=self.es_context.get_index_for_date(four_weeks_ago),
+            mappings=mappings,
+        )
+
+        es_storage = ESCrashStorage(config=self.config)
+
+        # Create a crash for this week and save it
+        now_uuid = "00000000-0000-0000-0000-000000120408"
+        raw_crash = {
+            "BuildID": "20200506000000",
+        }
+        processed_crash = {
+            field: "this week",
+            "date_processed": date_to_string(now),
```

Comment:
The ESCrashStorage figures out which index to put the crash in based on the `processed_crash["date_processed"]` field which is added by the processor.